src/main/resources/assets/computercraft/lua/rom/modules/main/http/pastebin.lua

```diff
@@ -0,0 +1,150 @@
+
+--- Attempts to guess the pastebin ID from the given code or URL
+local function extractId(paste)
+  local patterns = {
+    "^([%a%d]+)$",
+    "^https?://pastebin.com/([%a%d]+)$",
+    "^pastebin.com/([%a%d]+)$",
+    "^https?://pastebin.com/raw/([%a%d]+)$",
+    "^pastebin.com/raw/([%a%d]+)$",
+  }
+
+  for i = 1, #patterns do
+    local code = paste:match( patterns[i] )
+    if code then return code end
+  end
+
+  return nil
+end
+
+function download(url)
+  if type( url ) ~= "string" then
+    error( "bad argument #1 (expected string, got " .. type( url ) .. ")", 2 )
+  end
+
+  if not http then
+    return false, "Pastebin requires http API"
+  end
+
+  local paste = extractId( url )
+  if not paste then
+    return false, "Invalid pastebin code. The code is the ID at the end of the pastebin.com URL."
+  end
+
+  -- Add a cache buster so that spam protection is re-checked
+  local cacheBuster = ("%x"):format(math.random(0, 2^30))
+  local response, err = http.get(
+    "https://pastebin.com/raw/"..textutils.urlEncode( paste ).."?cb="..cacheBuster
+  )
+
+  if not response then
+    return response, err
+  end
+
+  -- If spam protection is activated, we get redirected to /paste with Content-Type: text/html
+  local headers = response.getResponseHeaders()
+  if not headers["Content-Type"] or not headers["Content-Type"]:find( "^text/plain" ) then
+    return false, "Pastebin blocked the download due to spam protection. Please complete the captcha in a web browser: https://pastebin.com/" .. textutils.urlEncode( paste )
```

Comment:
Still would like to come up with a better way of dealing with this. Error messages from API functions shouldn't really be this informative.